src/main/java/com/ms3_inc/camel/extensions/rest/SwaggerRequestValidator.java

```diff
@@ -16,91 +16,143 @@
+import org.openapi4j.operation.validator.validation.RequestValidator;
+import org.openapi4j.parser.OpenApi3Parser;
+import org.openapi4j.parser.model.v3.OpenApi3;
+import org.openapi4j.parser.model.v3.Server;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.util.MultiValueMap;
-import org.springframework.web.util.UriComponentsBuilder;
+import org.springframework.core.io.ClassPathResource;
 
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
 public class SwaggerRequestValidator extends AsyncProcessorSupport {
-    private final Logger LOGGER = LoggerFactory.getLogger(getClass());
-    private final OpenApiInteractionValidator validator;
+    private final Logger LOGGER = LoggerFactory.getLogger(SwaggerRequestValidator.class);
+    private final RequestValidator openapi4jValidator;
+    private OpenApi3 api;
 
     public SwaggerRequestValidator(String specPath) {
-        validator = OpenApiInteractionValidator
-                .createFor(specPath)
-                .build();
+        api = null;
+
+        try {
+            api = new OpenApi3Parser()
+                    .parse(new ClassPathResource(specPath).getFile(), false);
+        } catch (ResolutionException re) {
+            LOGGER.error(re.getMessage());
+        } catch (ValidationException ve) {
+            LOGGER.error(ve.getMessage());
+        } catch (IOException ioe) {
+            LOGGER.error(ioe.getMessage());
+        }
+
+        openapi4jValidator = new RequestValidator(api);
+    }
+
+    public SwaggerRequestValidator(String specPath, String basePath) {
+        api = null;
+
+        try {
+            api = new OpenApi3Parser()
+                    .parse(new ClassPathResource(specPath).getFile(), false);
+        } catch (ResolutionException re) {
+            LOGGER.error(re.getMessage());
+        } catch (ValidationException ve) {
+            LOGGER.error(ve.getMessage());
+        } catch (IOException ioe) {
+            LOGGER.error(ioe.getMessage());
+        }
+
+        api.setServers(new ArrayList<Server>(){
+            {
+                add(0, new Server().setUrl(basePath));
+            }
+        });
+
+        openapi4jValidator = new RequestValidator(api);
     }
 
     @Override
     public boolean process(Exchange exchange, AsyncCallback callback) {
         LOGGER.debug("Trying validation");
 
         try {
-            ValidationReport report = validator.validateRequest(fromExchange(exchange));
-            if (report.hasErrors()) {
-                LOGGER.debug(report.toString());
-                exchange.setException(new BadRequestException(fromReport(report)));
-            }
-
-            LOGGER.debug("Validating complete");
-        } catch (Exception ex) {
-            exchange.setException(new CamelException(ex));
+            openapi4jValidator.validate(fromExchange(exchange));
+        } catch (ValidationException e) {
```

Comment:
Is this the only Exception that this library will throw?

Reply:
Yes, as far as I can tell. [Here's the class] (https://github.com/openapi4j/openapi4j/blob/master/openapi-operation-validator/src/main/java/org/openapi4j/operation/validator/validation/RequestValidator.java)